upstairs/src/lib.rs

```diff
@@ -1347,6 +1397,20 @@ impl Upstairs {
         })
     }
 
+    fn set_active(&self) {
+        let mut active = self.active.lock().unwrap();
+        *active = true;
+    }
+
+    fn set_inactive(&self) {
```

Comment:
> If one downstairs receives this, do we assume the rest of the downstairs are going down also?

Yep, otherwise split brain can occur.

> Is the overall idea that the upstairs is not coming back once we have gone to NoLongerActive?

Yeah, the Upstairs should shut itself off.